[WIP] Changes to enable channel wise quant.

kimishpatel opened the pull request:
way.

Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

kimishpatel committed:
Changes to enable channel wise quant.
way.

Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

kostmo commented:
## :pill: CircleCI build failures summary and remediations
As of commit 4a60cf45:


* **9/9** failures introduced in this PR


## Detailed failure analysis

One may explore the probable reasons each build failed interactively [on the Dr. CI website](https://dr.pytorch.org/commit-details.html?sha1=4a60cf45d241bc98382f2a9391693c9bc40af157).

### :detective: 5 new failures recognized by patterns
The following build failures do not appear to be due to upstream breakage:
#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/4305043) pytorch_linux_xenial_py3_clang5_mobile_build (1/5)
**Step:** "Build" ([full log](https://dr.pytorch.org/api/view-log-full?build_id=88911367) | [pattern match details](https://dr.pytorch.org/build-details.html?build_id=88911367))
<details>
<summary>
<code>eluFused = true; torch::List<long int> = c10::List<long int>; int64_t = long int]::<lambda(int64_t)>]', declared using local type 'at::native::{anonymous}::QConvInt8<kSpatialDim, kReluFused>::QnnpackConv(at::Tensor, at::Tensor, torch::List<long int>, torch::List<long int>, torch::List<long int>, int64_t, double, int64_t) [with int kSpatialDim = 3; bool kReluFused = true; torch::List<long int> = c10::List<long int>; int64_t = long int]::<lambda(int64_t)>', is used but never defined [-fpermissive]</code>
</summary>

```
Jan 21 17:46:35 /var/lib/jenkins/workspace/aten/src/ATen/native/quantized/cpu/qnnpack/include/conv_utils.h:12:8: note: candidate: constexpr qnnpack::conv_param_t::conv_param_t(qnnpack::conv_param_t&&) 
Jan 21 17:46:35 /var/lib/jenkins/workspace/aten/src/ATen/native/quantized/cpu/qnnpack/include/conv_utils.h:12:8: note:   candidate expects 1 argument, 11 provided 
Jan 21 17:46:35 In file included from /usr/include/c++/5/algorithm:62:0, 
Jan 21 17:46:35                  from /var/lib/jenkins/workspace/aten/src/ATen/native/quantized/cpu/qconv.cpp:1: 
Jan 21 17:46:35 /usr/include/c++/5/bits/stl_algo.h: At global scope: 
Fused = false; torch::List<long int> = c10::List<long int>; int64_t = long int]::<lambda(int64_t)>]', declared using local type 'at::native::{anonymous}::QConvInt8<kSpatialDim, kReluFused>::QnnpackConv(at::Tensor, at::Tensor, torch::List<long int>, torch::List<long int>, torch::List<long int>, int64_t, double, int64_t) [with int kSpatialDim = 2; bool kReluFused = false; torch::List<long int> = c10::List<long int>; int64_t = long int]::<lambda(int64_t)>', is used but never defined [-fpermissive] 
Jan 21 17:46:35      all_of(_InputIterator __first, _InputIterator __last, _Predicate __pred) 
Jan 21 17:46:35      ^ 
luFused = true; torch::List<long int> = c10::List<long int>; int64_t = long int]::<lambda(int64_t)>]', declared using local type 'at::native::{anonymous}::QConvInt8<kSpatialDim, kReluFused>::QnnpackConv(at::Tensor, at::Tensor, torch::List<long int>, torch::List<long int>, torch::List<long int>, int64_t, double, int64_t) [with int kSpatialDim = 2; bool kReluFused = true; torch::List<long int> = c10::List<long int>; int64_t = long int]::<lambda(int64_t)>', is used but never defined [-fpermissive] 
Fused = false; torch::List<long int> = c10::List<long int>; int64_t = long int]::<lambda(int64_t)>]', declared using local type 'at::native::{anonymous}::QConvInt8<kSpatialDim, kReluFused>::QnnpackConv(at::Tensor, at::Tensor, torch::List<long int>, torch::List<long int>, torch::List<long int>, int64_t, double, int64_t) [with int kSpatialDim = 3; bool kReluFused = false; torch::List<long int> = c10::List<long int>; int64_t = long int]::<lambda(int64_t)>', is used but never defined [-fpermissive] 
luFused = true; torch::List<long int> = c10::List<long int>; int64_t = long int]::<lambda(int64_t)>]', declared using local type 'at::native::{anonymous}::QConvInt8<kSpatialDim, kReluFused>::QnnpackConv(at::Tensor, at::Tensor, torch::List<long int>, torch::List<long int>, torch::List<long int>, int64_t, double, int64_t) [with int kSpatialDim = 3; bool kReluFused = true; torch::List<long int> = c10::List<long int>; int64_t = long int]::<lambda(int64_t)>', is used but never defined [-fpermissive] 
Jan 21 17:46:35 make[2]: *** [caffe2/CMakeFiles/torch_cpu.dir/__/aten/src/ATen/native/quantized/cpu/qconv.cpp.o] Error 1 
Jan 21 17:46:35 make[1]: *** [caffe2/CMakeFiles/torch_cpu.dir/all] Error 2 
Jan 21 17:46:35 CMakeFiles/Makefile2:903: recipe for target 'caffe2/CMakeFiles/torch_cpu.dir/all' failed 
Jan 21 17:46:35 make: *** [all] Error 2 
Jan 21 17:46:35 Makefile:138: recipe for target 'all' failed 
Jan 21 17:46:35 + sccache_epilogue 
Jan 21 17:46:35 + echo '=================== sccache compilation log ===================' 
Jan 21 17:46:35 + python /var/lib/jenkins/workspace/.jenkins/pytorch/print_sccache_log.py /var/lib/jenkins/sccache_error.log 
Jan 21 17:46:35 =================== sccache compilation log =================== 
ument, 11 provided\n\u{1b}[01m\u{1b}[K/var/lib/jenkins/workspace/aten/src/ATen/native/quantized/cpu/qnnpack/include/conv_utils.h:12:8:\u{1b}[m\u{1b}[K \u{1b}[01;36m\u{1b}[Knote: \u{1b}[m\u{1b}[Kcandidate: constexpr qnnpack::conv_param_t::conv_param_t(qnnpack::conv_param_t&&)\n\u{1b}[01m\u{1b}[K/var/lib/jenkins/workspace/aten/src/ATen/native/quantized/cpu/qnnpack/include/conv_utils.h:12:8:\u{1b}[m\u{1b}[K \u{1b}[01;36m\u{1b}[Knote: \u{1b}[m\u{1b}[K  candidate expects 1 argument, 11 provided\n" } 
```

</details>

#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/4305035) caffe2_onnx_py2_gcc5_ubuntu16_04_build (2/5)
**Step:** "Build" ([full log](https://dr.pytorch.org/api/view-log-full?build_id=88911396) | [pattern match details](https://dr.pytorch.org/build-details.html?build_id=88911396))
<details>
<summary>
<code>eluFused = true; torch::List<long int> = c10::List<long int>; int64_t = long int]::<lambda(int64_t)>]', declared using local type 'at::native::{anonymous}::QConvInt8<kSpatialDim, kReluFused>::QnnpackConv(at::Tensor, at::Tensor, torch::List<long int>, torch::List<long int>, torch::List<long int>, int64_t, double, int64_t) [with int kSpatialDim = 3; bool kReluFused = true; torch::List<long int> = c10::List<long int>; int64_t = long int]::<lambda(int64_t)>', is used but never defined [-fpermissive]</code>
</summary>

```
Jan 21 17:47:06 /var/lib/jenkins/workspace/aten/src/ATen/native/quantized/cpu/qnnpack/include/conv_utils.h:12:8: note: candidate: constexpr qnnpack::conv_param_t::conv_param_t(qnnpack::conv_param_t&&) 
Jan 21 17:47:06 /var/lib/jenkins/workspace/aten/src/ATen/native/quantized/cpu/qnnpack/include/conv_utils.h:12:8: note:   candidate expects 1 argument, 11 provided 
Jan 21 17:47:06 In file included from /usr/include/c++/5/algorithm:62:0, 
Jan 21 17:47:06                  from /var/lib/jenkins/workspace/aten/src/ATen/native/quantized/cpu/qconv.cpp:1: 
Jan 21 17:47:06 /usr/include/c++/5/bits/stl_algo.h: At global scope: 
Fused = false; torch::List<long int> = c10::List<long int>; int64_t = long int]::<lambda(int64_t)>]', declared using local type 'at::native::{anonymous}::QConvInt8<kSpatialDim, kReluFused>::QnnpackConv(at::Tensor, at::Tensor, torch::List<long int>, torch::List<long int>, torch::List<long int>, int64_t, double, int64_t) [with int kSpatialDim = 2; bool kReluFused = false; torch::List<long int> = c10::List<long int>; int64_t = long int]::<lambda(int64_t)>', is used but never defined [-fpermissive] 
Jan 21 17:47:06      all_of(_InputIterator __first, _InputIterator __last, _Predicate __pred) 
Jan 21 17:47:06      ^ 
luFused = true; torch::List<long int> = c10::List<long int>; int64_t = long int]::<lambda(int64_t)>]', declared using local type 'at::native::{anonymous}::QConvInt8<kSpatialDim, kReluFused>::QnnpackConv(at::Tensor, at::Tensor, torch::List<long int>, torch::List<long int>, torch::List<long int>, int64_t, double, int64_t) [with int kSpatialDim = 2; bool kReluFused = true; torch::List<long int> = c10::List<long int>; int64_t = long int]::<lambda(int64_t)>', is used but never defined [-fpermissive] 
Fused = false; torch::List<long int> = c10::List<long int>; int64_t = long int]::<lambda(int64_t)>]', declared using local type 'at::native::{anonymous}::QConvInt8<kSpatialDim, kReluFused>::QnnpackConv(at::Tensor, at::Tensor, torch::List<long int>, torch::List<long int>, torch::List<long int>, int64_t, double, int64_t) [with int kSpatialDim = 3; bool kReluFused = false; torch::List<long int> = c10::List<long int>; int64_t = long int]::<lambda(int64_t)>', is used but never defined [-fpermissive] 
luFused = true; torch::List<long int> = c10::List<long int>; int64_t = long int]::<lambda(int64_t)>]', declared using local type 'at::native::{anonymous}::QConvInt8<kSpatialDim, kReluFused>::QnnpackConv(at::Tensor, at::Tensor, torch::List<long int>, torch::List<long int>, torch::List<long int>, int64_t, double, int64_t) [with int kSpatialDim = 3; bool kReluFused = true; torch::List<long int> = c10::List<long int>; int64_t = long int]::<lambda(int64_t)>', is used but never defined [-fpermissive] 
Jan 21 17:47:06 make[2]: *** [caffe2/CMakeFiles/torch_cpu.dir/__/aten/src/ATen/native/quantized/cpu/qconv.cpp.o] Error 1 
Jan 21 17:47:06 caffe2/CMakeFiles/torch_cpu.dir/build.make:4138: recipe for target 'caffe2/CMakeFiles/torch_cpu.dir/__/aten/src/ATen/native/quantized/cpu/qconv.cpp.o' failed 
Jan 21 17:47:07 CMakeFiles/Makefile2:5487: recipe for target 'caffe2/CMakeFiles/torch_cpu.dir/all' failed 
Jan 21 17:47:07 make[1]: *** [caffe2/CMakeFiles/torch_cpu.dir/all] Error 2 
Jan 21 17:47:07 make: *** [all] Error 2 
Jan 21 17:47:07 Makefile:138: recipe for target 'all' failed 
Jan 21 17:47:07 Traceback (most recent call last): 
Jan 21 17:47:07   File "setup.py", line 737, in <module> 
Jan 21 17:47:07 Building wheel torch-1.5.0a0+4a60cf4 
Jan 21 17:47:07 -- Building version 1.5.0a0+4a60cf4 
```

</details>

#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/4305048) pytorch_linux_xenial_py2_7_9_build (3/5)
**Step:** "Build" ([full log](https://dr.pytorch.org/api/view-log-full?build_id=88911412) | [pattern match details](https://dr.pytorch.org/build-details.html?build_id=88911412))
<details>
<summary>
<code>Jan 21 17:47:13 /usr/include/c++/7/bits/stl_algo.h:508:5: error: 'bool std::all_of(_IIter, _IIter, _Predicate) [with _IIter = const long int*; _Predicate = at::native::{anonymous}::QConvInt8<kSpatialDim, kReluFused>::QnnpackConv(at::Tensor, at::Tensor, torch::List<long int>, torch::List<long int>, torch::List<long int>, int64_t, double, int64_t) [with int kSpatialDim = 2; bool kReluFused = false]::<lambda(int64_t)>]' used but never defined [-Werror]</code>
</summary>

```
Jan 21 17:47:13 In file included from /usr/include/c++/7/algorithm:62:0, 
Jan 21 17:47:13                  from /var/lib/jenkins/workspace/aten/src/ATen/native/quantized/cpu/qconv.cpp:1: 
Jan 21 17:47:13 /usr/include/c++/7/bits/stl_algo.h: At global scope: 
Fused = false; torch::List<long int> = c10::List<long int>; int64_t = long int]::<lambda(int64_t)>]', declared using local type 'at::native::{anonymous}::QConvInt8<kSpatialDim, kReluFused>::QnnpackConv(at::Tensor, at::Tensor, torch::List<long int>, torch::List<long int>, torch::List<long int>, int64_t, double, int64_t) [with int kSpatialDim = 2; bool kReluFused = false; torch::List<long int> = c10::List<long int>; int64_t = long int]::<lambda(int64_t)>', is used but never defined [-fpermissive] 
Jan 21 17:47:13      all_of(_InputIterator __first, _InputIterator __last, _Predicate __pred) 
Jan 21 17:47:13      ^~~~~~ 
luFused = true; torch::List<long int> = c10::List<long int>; int64_t = long int]::<lambda(int64_t)>]', declared using local type 'at::native::{anonymous}::QConvInt8<kSpatialDim, kReluFused>::QnnpackConv(at::Tensor, at::Tensor, torch::List<long int>, torch::List<long int>, torch::List<long int>, int64_t, double, int64_t) [with int kSpatialDim = 2; bool kReluFused = true; torch::List<long int> = c10::List<long int>; int64_t = long int]::<lambda(int64_t)>', is used but never defined [-fpermissive] 
Fused = false; torch::List<long int> = c10::List<long int>; int64_t = long int]::<lambda(int64_t)>]', declared using local type 'at::native::{anonymous}::QConvInt8<kSpatialDim, kReluFused>::QnnpackConv(at::Tensor, at::Tensor, torch::List<long int>, torch::List<long int>, torch::List<long int>, int64_t, double, int64_t) [with int kSpatialDim = 3; bool kReluFused = false; torch::List<long int> = c10::List<long int>; int64_t = long int]::<lambda(int64_t)>', is used but never defined [-fpermissive] 
luFused = true; torch::List<long int> = c10::List<long int>; int64_t = long int]::<lambda(int64_t)>]', declared using local type 'at::native::{anonymous}::QConvInt8<kSpatialDim, kReluFused>::QnnpackConv(at::Tensor, at::Tensor, torch::List<long int>, torch::List<long int>, torch::List<long int>, int64_t, double, int64_t) [with int kSpatialDim = 3; bool kReluFused = true; torch::List<long int> = c10::List<long int>; int64_t = long int]::<lambda(int64_t)>', is used but never defined [-fpermissive] 
Jan 21 17:47:13 /usr/include/c++/7/bits/stl_algo.h:508:5: error: 'bool std::all_of(_IIter, _IIter, _Predicate) [with _IIter = const long int*; _Predicate = at::native::{anonymous}::QConvInt8<kSpatialDim, kReluFused>::QnnpackConv(at::Tensor, at::Tensor, torch::List<long int>, torch::List<long int>, torch::List<long int>, int64_t, double, int64_t) [with int kSpatialDim = 2; bool kReluFused = true]::<lambda(int64_t)>]' used but never defined [-Werror] 
Jan 21 17:47:13 /usr/include/c++/7/bits/stl_algo.h:508:5: error: 'bool std::all_of(_IIter, _IIter, _Predicate) [with _IIter = const long int*; _Predicate = at::native::{anonymous}::QConvInt8<kSpatialDim, kReluFused>::QnnpackConv(at::Tensor, at::Tensor, torch::List<long int>, torch::List<long int>, torch::List<long int>, int64_t, double, int64_t) [with int kSpatialDim = 2; bool kReluFused = false]::<lambda(int64_t)>]' used but never defined [-Werror] 
Jan 21 17:47:13 cc1plus: all warnings being treated as errors 
Jan 21 17:47:13 make[2]: *** [caffe2/CMakeFiles/torch_cpu.dir/__/aten/src/ATen/native/quantized/cpu/qconv.cpp.o] Error 1 
Jan 21 17:47:13 caffe2/CMakeFiles/torch_cpu.dir/build.make:4138: recipe for target 'caffe2/CMakeFiles/torch_cpu.dir/__/aten/src/ATen/native/quantized/cpu/qconv.cpp.o' failed 
Jan 21 17:47:21 make[1]: *** [caffe2/CMakeFiles/torch_cpu.dir/all] Error 2 
Jan 21 17:47:21 CMakeFiles/Makefile2:5337: recipe for target 'caffe2/CMakeFiles/torch_cpu.dir/all' failed 
Jan 21 17:47:21 make: *** [all] Error 2 
Jan 21 17:47:21 Makefile:138: recipe for target 'all' failed 
Jan 21 17:47:21 Building wheel torch-1.5.0a0+4a60cf4 
Jan 21 17:47:21 -- Building version 1.5.0a0+4a60cf4 
_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX=/var/lib/jenkins/workspace/torch -DCMAKE_PREFIX_PATH=/opt/python/2.7.9/lib/python2.7/site-packages -DNUMPY_INCLUDE_DIR=/opt/python/2.7.9/lib/python2.7/site-packages/numpy/core/include -DPYTHON_EXECUTABLE=/opt/python/2.7.9/bin/python -DPYTHON_INCLUDE_DIR=/opt/python/2.7.9/include/python2.7 -DPYTHON_LIBRARY=/opt/python/2.7.9/lib/libpython2.7.so.1.0 -DTORCH_BUILD_VERSION=1.5.0a0+4a60cf4 -DUSE_MKLDNN=1 -DUSE_NUMPY=True -DWERROR=1 /var/lib/jenkins/workspace 
```

</details>

#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/4305032) caffe2_onnx_py3_6_clang7_ubuntu16_04_build (4/5)
**Step:** "Build" ([full log](https://dr.pytorch.org/api/view-log-full?build_id=88911429) | [pattern match details](https://dr.pytorch.org/build-details.html?build_id=88911429))
<details>
<summary>
<code>Jan 21 17:47:41 /var/lib/jenkins/workspace/aten/src/ATen/native/quantized/cpu/qconv.cpp:551:27: error: no matching constructor for initialization of 'qnnpack::conv_param_t'</code>
</summary>

```
Jan 21 17:47:38 /var/lib/jenkins/workspace/aten/src/ATen/core/op_registration/op_registration.h:150:25: note: in instantiation of function template specialization 'c10::KernelFunction::makeFromUnboxedFunctorFactory<at::native::(anonymous namespace)::QConvPackWeightInt8<3>, false>' requested here 
Jan 21 17:47:38         KernelFunction::makeFromUnboxedFunctorFactory<KernelFunctor>(detail::KernelFactory<KernelFunctor, std::decay_t<ConstructorParameters>...>(std::forward<ConstructorParameters>(constructorParameters)...)), 
Jan 21 17:47:38                         ^ 
Jan 21 17:47:38 /var/lib/jenkins/workspace/aten/src/ATen/native/quantized/cpu/qconv_prepack.cpp:315:47: note: in instantiation of function template specialization 'c10::RegisterOperators::Options::kernel<at::native::(anonymous namespace)::QConvPackWeightInt8<3>>' requested here 
Jan 21 17:47:38             c10::RegisterOperators::options().kernel<QConvPackWeightInt8<3>>( 
Jan 21 17:47:38                                               ^ 
Jan 21 17:47:38 3 warnings and 1 error generated. 
Jan 21 17:47:38 caffe2/CMakeFiles/torch_cpu.dir/build.make:4090: recipe for target 'caffe2/CMakeFiles/torch_cpu.dir/__/aten/src/ATen/native/quantized/cpu/qconv_prepack.cpp.o' failed 
Jan 21 17:47:38 make[2]: *** [caffe2/CMakeFiles/torch_cpu.dir/__/aten/src/ATen/native/quantized/cpu/qconv_prepack.cpp.o] Error 1 
Jan 21 17:47:38 make[2]: *** Waiting for unfinished jobs.... 
Jan 21 17:47:41 /var/lib/jenkins/workspace/aten/src/ATen/native/quantized/cpu/qconv.cpp:551:27: error: no matching constructor for initialization of 'qnnpack::conv_param_t' 
Jan 21 17:47:41     qnnpack::conv_param_t conv_p( 
Jan 21 17:47:41                           ^ 
Jan 21 17:47:41 /var/lib/jenkins/workspace/aten/src/ATen/native/quantized/cpu/qnnpack/include/conv_utils.h:33:3: note: candidate constructor not viable: no known conversion from 'const long' to 'const uint8_t *' (aka 'const unsigned char *') for 8th argument 
Jan 21 17:47:41   conv_param_t( 
Jan 21 17:47:41   ^ 
Jan 21 17:47:41 /var/lib/jenkins/workspace/aten/src/ATen/native/quantized/cpu/qnnpack/include/conv_utils.h:12:8: note: candidate constructor (the implicit copy constructor) not viable: requires 1 argument, but 11 were provided 
Jan 21 17:47:41 struct conv_param_t { 
Jan 21 17:47:41        ^ 
Jan 21 17:47:41 /var/lib/jenkins/workspace/aten/src/ATen/native/quantized/cpu/qnnpack/include/conv_utils.h:12:8: note: candidate constructor (the implicit move constructor) not viable: requires 1 argument, but 11 were provided 
Jan 21 17:47:41 1 error generated. 
```

</details>

#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/4305045) pytorch_macos_10_13_py3_build (5/5)
**Step:** "Build" ([full log](https://dr.pytorch.org/api/view-log-full?build_id=88911464) | [pattern match details](https://dr.pytorch.org/build-details.html?build_id=88911464))
<details>
<summary>
<code>Jan 21 09:48:35 FAILED: caffe2/CMakeFiles/torch_cpu.dir/__/aten/src/ATen/native/quantized/cpu/qconv_prepack.cpp.o</code>
</summary>

```
Jan 21 09:48:33                           ^ 
Jan 21 09:48:33 ../aten/src/ATen/native/quantized/cpu/qnnpack/include/conv_utils.h:33:3: note: candidate constructor not viable: no known conversion from 'const long long' to 'const uint8_t *' (aka 'const unsigned char *') for 8th argument 
Jan 21 09:48:33   conv_param_t( 
Jan 21 09:48:33   ^ 
Jan 21 09:48:33 ../aten/src/ATen/native/quantized/cpu/qnnpack/include/conv_utils.h:12:8: note: candidate constructor (the implicit move constructor) not viable: requires 1 argument, but 11 were provided 
Jan 21 09:48:33 struct conv_param_t { 
Jan 21 09:48:33        ^ 
Jan 21 09:48:33 ../aten/src/ATen/native/quantized/cpu/qnnpack/include/conv_utils.h:12:8: note: candidate constructor (the implicit copy constructor) not viable: requires 1 argument, but 11 were provided 
Jan 21 09:48:33 1 warning and 1 error generated. 
Jan 21 09:48:35 [1790/2997] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/aten/src/ATen/native/quantized/cpu/qconv_prepack.cpp.o 
Jan 21 09:48:35 FAILED: caffe2/CMakeFiles/torch_cpu.dir/__/aten/src/ATen/native/quantized/cpu/qconv_prepack.cpp.o  
eld-initializers -Wno-write-strings -Wno-unknown-pragmas -Wno-missing-braces -fvisibility=hidden -O2 -DCAFFE2_BUILD_MAIN_LIB -DASMJIT_STATIC -std=gnu++14 -MD -MT caffe2/CMakeFiles/torch_cpu.dir/__/aten/src/ATen/native/quantized/cpu/qconv_prepack.cpp.o -MF caffe2/CMakeFiles/torch_cpu.dir/__/aten/src/ATen/native/quantized/cpu/qconv_prepack.cpp.o.d -o caffe2/CMakeFiles/torch_cpu.dir/__/aten/src/ATen/native/quantized/cpu/qconv_prepack.cpp.o -c ../aten/src/ATen/native/quantized/cpu/qconv_prepack.cpp 
Jan 21 09:48:35 In file included from ../aten/src/ATen/native/quantized/cpu/qconv_prepack.cpp:7: 
Jan 21 09:48:35 In file included from ../aten/src/ATen/native/quantized/cpu/fbgemm_utils.h:5: 
Jan 21 09:48:35 ../third_party/fbgemm/include/fbgemm/QuantUtils.h:122:29: warning: comparison of integers of different signs: 'std::size_t' (aka 'unsigned long') and 'int' [-Wsign-compare] 
Jan 21 09:48:35   for (std::size_t i = 0; i < len; i++) { 
Jan 21 09:48:35                           ~ ^ ~~~ 
Jan 21 09:48:35 ../aten/src/ATen/native/quantized/cpu/qconv_prepack.cpp:270:27: error: no matching constructor for initialization of 'qnnpack::conv_param_t' 
Jan 21 09:48:35     qnnpack::conv_param_t conv_p( 
Jan 21 09:48:35                           ^ 
Jan 21 09:48:35 ../aten/src/ATen/native/quantized/cpu/qnnpack/include/conv_utils.h:33:3: note: candidate constructor not viable: no known conversion from 'int64_t' (aka 'long long') to 'const uint8_t *' (aka 'const unsigned char *') for 8th argument 
```

</details>



---

<details><summary>This comment was automatically generated by <a href="https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment">Dr. CI</a> (expand for details).</summary>Follow <a href="https://dr.pytorch.org/admin/comments-opt-out.html">this link to opt-out</a> of these comments for your Pull Requests.

Please report bugs/suggestions on the <a href="https://github.com/kostmo/circleci-failure-tracker/issues">GitHub issue tracker</a>.
</details>

kimishpatel committed:
Enable per channel zero point

Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

kimishpatel committed:
Make deconvolution channelwise quant changes.

Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

kimishpatel committed:
Fixed failing tests of depthwise convs.
Note that we do not do channel wise quant for depthwaise conv.
1. It does not make much sense due to each kernel having single input
channel.
2. It will add too much perf overhead to the kernel.

This can be revisited later.

Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

kimishpatel committed:
Fixed some bugs.
Specifically.
1. Pushing 5 or 11 registers to stack seems to seg fault.
2. Now that we changed the struct of params, compiler assigns members to
allocated memory differently than expected. This required use to change
how the pointer is incremented.

Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

kimishpatel committed:
Fixed a bug for 64 bit version asm.

Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

kimishpatel committed:
First step towards adding support for per channel requantization scale.

Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

kimishpatel committed:
Modified the kernels to support channel wise quant.

Summary:
In the process changes SSE2 implementation to be little simpler assuming
we care less for perf on those for mobile builds.
Due to rounding issues, sometimes abs difference between reference and
under test implementation is 1. Fixed tests to check if the difference
is larger than 1. Fail only if so.

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:
